pkg/controller/consoleservice/consoleservice_controller.go

```diff
@@ -463,39 +463,44 @@ func applyService(consoleService *v1beta1.ConsoleService, service *corev1.Servic
 
 func (r *ReconcileConsoleService) reconcilePrometheusRule(ctx context.Context, consoleservice *v1beta1.ConsoleService) (reconcile.Result, error) {
 
-	if !util.HasApi(monitoringGVK) {
-		return reconcile.Result{}, nil
-	}
+	monitoringEnabled := os.Getenv("ENABLE_MONITORING")
 
-	prometheusRule := &monitoringv1.PrometheusRule{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "enmasse-console-rules",
-			Namespace: consoleservice.Namespace,
-		},
-	}
+	if monitoringEnabled == "true" {
 
-	_, err := controllerutil.CreateOrUpdate(ctx, r.client, prometheusRule, func() error {
+		if !util.HasApi(monitoringGVK) {
+			return reconcile.Result{}, nil
+		}
+
+		prometheusRule := &monitoringv1.PrometheusRule{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "enmasse-console-rules",
+				Namespace: consoleservice.Namespace,
+			},
+		}
 
-		install.ApplyDefaultLabels(&prometheusRule.ObjectMeta, "consoleservice", consoleservice.Name)
-		install.ApplyCustomLabel(&prometheusRule.ObjectMeta, "monitoring-key", "middleware")
-		prometheusRule.Spec = monitoringv1.PrometheusRuleSpec{
-			Groups: []monitoringv1.RuleGroup{
-				{
-					Name: "ComponentHealth",
-					Rules: []monitoringv1.Rule{
-						{
-							Record: "enmasse_component_health",
-							Expr:   intstr.FromString("up{job='console',namespace='" + util.GetEnvOrDefault("NAMESPACE", "enmasse-infra") + "'} or on(namespace) (1- absent(up{job='console',namespace='" + util.GetEnvOrDefault("NAMESPACE", "enmasse-infra") + "'}))"),
+		_, err := controllerutil.CreateOrUpdate(ctx, r.client, prometheusRule, func() error {
```

Comment:
If a user were to enable `ENABLE_MONITORING` then later to disable it again, I think the system should revert to its original state i.e. the PrometheusRule object should be removed.  Should there be an if monitoringEnabled { } else {....} clause, where we attempt gracefully to tidy away what was created?

Reply:
Creating a separate issue for this as there are a number of monitoring resources installed in different places